catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorTest.java

```diff
@@ -22,322 +22,196 @@
-        // second
-        // route that had this change in it. (So still get NPE because first route fires and cannot
-        // resolve the "content" scheme)
-        camelContext.getRouteDefinitions()
-                .get(0)
-                .adviceWith(camelContext, new AdviceWithRouteBuilder() {
-                    @Override
-                    public void configure() throws Exception {
-                        // weave the node in the route which has id = content://framework
-                        // and replace it with the following route path
-                        weaveByToString(".*content:framework.*").replace()
-                                .to("mock:result");
+    private void doAndVerifyFileMove(File destinationFolder, File monitoredFolder,
+            String inputFileName) throws Exception {
+        doFileMove(destinationFolder, inputFileName);
+        Failsafe.with(new RetryPolicy().withMaxRetries(10)
+                .withDelay(1, TimeUnit.SECONDS))
+                .run((executionContext) -> {
+                    LOGGER.debug("Veryifying the file moved, retry {}",
+                            executionContext.getExecutions());
+                    if (!verifyFileMovedToIngestedDirectory(monitoredFolder, inputFileName)) {
+                        throw new RuntimeException();
                     }
                 });
 
-        // Initial Camel route should now be created
-        List<RouteDefinition> routeDefinitions = contentDirectoryMonitor.getRouteDefinitions();
-        assertThat(routeDefinitions.size(), is(1));
-        LOGGER.debug("routeDefinition = {}", routeDefinitions.get(0));
-
-        return routeDefinitions.get(0);
+        assertTrue(verifyFileMovedToIngestedDirectory(monitoredFolder, inputFileName));
     }
 
-    private RouteDefinition createRoute(String monitoredDirectory, boolean copyIngestedFiles)
-            throws Exception {
-
-        // Simulates what container would do for <camel:camelContext id="camelContext">
-        // declaration in beans.xml file
-        camelContext = (ModelCamelContext) super.createCamelContext();
-        camelContext.start();
-
-        // Map the "content" scheme to a mock component so that we do not have to
-        // mock the entire custom ContentComponent and include its implementation
-        // in pom with scope=test
-        camelContext.addComponent("content", new MockComponent());
-
-        contentDirectoryMonitor = new ContentDirectoryMonitor(camelContext);
-        contentDirectoryMonitor.systemSubjectBinder = noOpProcessor;
-        contentDirectoryMonitor.setMonitoredDirectoryPath(monitoredDirectory);
-        contentDirectoryMonitor.setCopyIngestedFiles(copyIngestedFiles);
+    private void doAndVerifyFileDidNotMove(File destinationFolder, File monitoredFolder,
+            String inputFileName) throws Exception {
+        doFileMove(destinationFolder, inputFileName);
+        for (int i = 0; i < 5; i++) {
```

Comment:
In the above case, I am assuming the reasonable amount of time for a file to copy is 5 seconds. This pertains to the issue in the comment above; how are we able to predict the user's system spec's to arrive at reasonable wait values? 

Ideally we want to avoid the use of waiting entirely, but I don't see how that's possible in this scenario.